backend/core/src/auth/services/user.service.ts

```diff
@@ -621,7 +621,16 @@ export class UserService {
   }
 
   private static getPublicConfirmationUrl(appUrl: string, user: User) {
-    return `${appUrl}?token=${user.confirmationToken}`
+    const urlObj = new URL(appUrl)
```

Comment:
This looks very similar to the one used in the `email.service.ts` file below if you can I would say pull that out into a helper